tools/pull-request-labeler/src/utils/process-commit-messages.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { splitStringIntoCommitMessage } from './split-message-into-components';
+
+/**
+ *  Rules for cherrypicking labels for pull requests targeting master
+ *
+ *                         PR contains
+ *                    BREAKING CHANGE COMMITS?
+ *                    /               \
+ *                  /                  \
+ *                Yes                  No
+ *               /                      \
+ *             /                         \
+ * Add label *target:major*          contains *feat* commits
+ *                                     /            \
+ *                                   /               \
+ *                                Yes                No
+ *                                /                   \
+ *                              /                      \
+ *                       Add labels              contains *fix* or *perf* commits
+ *                       *target:minor*              /           \
+ *                                                 /              \
+ *                                              Yes               No
+ *                                              /                  \
+ *                                            /                     \
+ *                                       Add labels                Add labels
+ *                                       *target:patch*            *target:minor*
+ *                                       *target:minor*            *target:patch*
+ *
+ */
+export function processCommitMessages(
+  commitMessages: string[],
+): {
+  targets: string[];
+  errors: string[];
+} {
+  const targets: string[] = [];
+  const errors: string[] = [];
+  // Parse the commit messages and split them into their own components.
+  const parsedCommitMessages = commitMessages.map(commit =>
+    splitStringIntoCommitMessage(commit),
+  );
+
+  // Filter feature commits in the pull request.
+  const featureCommits = parsedCommitMessages.filter(
+    commit => commit.type === 'feat',
+  );
+  // Filter fix commits in the pull request.
+  const fixCommits = parsedCommitMessages.filter(
+    commit => commit.type === 'fix' || commit.type === 'perf',
```

Comment:
if you would use a const enum for the commit types you could reference them here CommitTypes.FIX and you would have the benefit of having typings and values at cones